server/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -624,9 +675,16 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
         if (source != null) {
             XContentHelper.writeRawField(SourceFieldMapper.NAME, source, builder, params);
         }
-        if (!otherFields.isEmpty()) {
-            builder.startObject(Fields.FIELDS);
```

Comment:
We can't break the json response like this. The union of the document and metadata fields should still exist under "fields".

Reply:
corrected. Looks like the metadata fields were just on the root object, while only document fields were under `fields`. Corrected to what it has been before. 

It is slightly concerning that we have a dependency to static `MapperService.getAllMetaFields()` in the  json deserialization part, which contradicts our strategic plan of separating the metadata and document fields in different maps on `SearchHit` level. This is not the objective of this PR, but not sure how can we deal with this in future